src/clientversion.cpp

```diff
@@ -99,11 +100,20 @@ std::string FormatSubVersion(const std::string& name, int nClientVersion, const
     std::ostringstream ss;
     ss << "/";
     ss << name << ":" << FormatVersion(nClientVersion);
-    if (!comments.empty())
+
+    const std::string key = "-uacomment";
```

Comment:
Right now, FormatSubVersion is a pure utility function that doesn't use any global state. I think it should stay that way.  E.g. by adding an `const std::vector<std::string> &commentList` argument. Then pass in this data at the caller side.

Also makes it easier to test this in `util_tests`.